### PR TITLE
Support both TPR and CRD with custom objects spec

### PR DESCRIPTION
**Pull request by mbohlool**
ref: https://github.com/kubernetes-incubator/client-python/pull/283
ref: https://github.com/kubernetes-incubator/client-python/issues/251

cc @djmckinley @enisoc

### CODE REVIEW

File: openapi/custom_objects_spec.json [hunk 1]
```diff
@@ -0,0 +1,560 @@
+{
+  "/apis/{group}/{version}/{kind}": {
```
Comment (enisoc) [inline openapi/custom_objects_spec.json]: `{kind}` should now be `{plural}` right?

Comment (mbohlool) [inline openapi/custom_objects_spec.json]: Right. Fixed.

**Comment by djmckinley** [comment]
I tried the list_cluster_custom_object() function to list third party resources in all namespaces, as you suggested.  However, it did not work.  Apparently it only lists objects that are defined by new v1.7 CRDs that are explicitly non-namespaced.  I don't have Kubernetes 1.7 installed, so I cannot verify that - but I can see that list_namespaced_custom_object() does work to list TPRs in a single namespace, but there doesn't seem to be a way to list namespaced TPRs in all namespaces.

I ran this to test it against a Kubernetes 1.6.2 system with a "Test" TPR defined, with four instances created, two in the 'default' namespace, and two in the 'mynamespace' namespace:

```
import kubernetes
from pprint import pprint
import sys

try:
    kubernetes.config.load_incluster_config()
    print( "configured in cluster with service account" )
except:
    try:
        kubernetes.config.load_kube_config()
        print( "configured via kubeconfig file" )
    except:
        print( "No Kubernetes configuration found" )
        sys.exit(1)

api_instance = kubernetes.client.CustomObjectsApi()

fqdn = 'occloud.oracle.com'
version = 'v1'
resource = 'tests'

print ("\nCalling list_namespaced_custom_object for 'default' namespace\n")

try:
    # List all Subnamespaces in default namespace
    api_response = api_instance.list_namespaced_custom_object( fqdn, version, 'default', resource )
    pprint(api_response)
except kubernetes.client.rest.ApiException as e:
    print("Exception when calling CustomObjectsApi.list_namespaced_custom_object: %s\n" % e)

print ("\nCalling list_namespaced_custom_object for 'mynamespace' namespace\n")

try:
    # List all Subnamespaces in default namespace
    api_response = api_instance.list_namespaced_custom_object( fqdn, version, 'mynamespace', resource )
    pprint(api_response)
except kubernetes.client.rest.ApiException as e:
    print("Exception when calling CustomObjectsApi.list_namespaced_custom_object: %s\n" % e)

print ("\nCalling list_cluster_custom_object\n")

try:
    # List all SubNamespaces
    api_response = api_instance.list_cluster_custom_object( fqdn, version, resource )
    pprint(api_response)
except kubernetes.client.rest.ApiException as e:
    print("Exception when calling CustomObjectsApi.list_cluster_custom_object: %s\n" % e)
```

Running this produced this output:

```
configured via kubeconfig file

Calling list_namespaced_custom_object for 'default' namespace

{'apiVersion': 'occloud.oracle.com/v1',
 'items': [{'apiVersion': 'occloud.oracle.com/v1',
            'kind': 'Test',
            'metadata': {'creationTimestamp': '2017-07-06T22:16:36Z',
                         'name': 'test1',
                         'namespace': 'default',
                         'resourceVersion': '2449709',
                         'selfLink': '/apis/occloud.oracle.com/v1/namespaces/default/tests/test1',
                         'uid': 'c699909a-6298-11e7-9dda-fa163e3667f1'},
            'spec': {'testCount': 1, 'testType': 'test1 type'}},
           {'apiVersion': 'occloud.oracle.com/v1',
            'kind': 'Test',
            'metadata': {'creationTimestamp': '2017-07-06T22:17:06Z',
                         'name': 'test2',
                         'namespace': 'default',
                         'resourceVersion': '2449765',
                         'selfLink': '/apis/occloud.oracle.com/v1/namespaces/default/tests/test2',
                         'uid': 'd8402752-6298-11e7-9dda-fa163e3667f1'},
            'spec': {'testCount': 2, 'testType': 'test2 type'}}],
 'kind': 'TestList',
 'metadata': {'resourceVersion': '2452325',
              'selfLink': '/apis/occloud.oracle.com/v1/namespaces/default/tests'}}

Calling list_namespaced_custom_object for 'mynamespace' namespace

{'apiVersion': 'occloud.oracle.com/v1',
 'items': [{'apiVersion': 'occloud.oracle.com/v1',
            'kind': 'Test',
            'metadata': {'creationTimestamp': '2017-07-06T22:37:56Z',
                         'name': 'test3',
                         'namespace': 'mynamespace',
                         'resourceVersion': '2452133',
                         'selfLink': '/apis/occloud.oracle.com/v1/namespaces/mynamespace/tests/test3',
                         'uid': 'c1b8fa7a-629b-11e7-9dda-fa163e3667f1'},
            'spec': {'testCount': 3, 'testType': 'test3 type'}},
           {'apiVersion': 'occloud.oracle.com/v1',
            'kind': 'Test',
            'metadata': {'creationTimestamp': '2017-07-06T22:37:59Z',
                         'name': 'test4',
                         'namespace': 'mynamespace',
                         'resourceVersion': '2452139',
                         'selfLink': '/apis/occloud.oracle.com/v1/namespaces/mynamespace/tests/test4',
                         'uid': 'c3973d62-629b-11e7-9dda-fa163e3667f1'},
            'spec': {'testCount': 4, 'testType': 'test4 type'}}],
 'kind': 'TestList',
 'metadata': {'resourceVersion': '2452325',
              'selfLink': '/apis/occloud.oracle.com/v1/namespaces/mynamespace/tests'}}

Calling list_cluster_custom_object

Exception when calling CustomObjectsApi.list_cluster_custom_object: (404)
Reason: Not Found
HTTP response headers: HTTPHeaderDict({'Date': 'Thu, 06 Jul 2017 22:39:41 GMT', 'Content-Type': 'application/json', 'Content-Length': '174'})
HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"the server could not find the requested resource","reason":"NotFound","details":{},"code":404}
```

**Comment by mbohlool** [comment]
@djmckinley do you know the `kubectl` command to get all TPRs in all namespaces? Can you run that command with `-v9` and paste the result here?

**Comment by enisoc** [comment]
> Apparently it only lists objects that are defined by new v1.7 CRDs that are explicitly non-namespaced.

I think for CRD, it should also work to list namespaced custom objects across all namespaces, according to my experiment below. Perhaps TPR just doesn't support this semantic. It's unlikely we would fix that for TPR since it's deprecated and the code is already deleted in the master branch.

It would be nice to verify this PR against CRD in a 1.7 cluster to make sure the generated client works the same as my experiment. I could do that, but I'd need some steps for how to get the client running.

```console
$ kubectl get --raw /apis/stable.example.com/v1/crontabs/ | json_pp
{
   "kind" : "CronTabList",
   "metadata" : {
      "resourceVersion" : "1805",
      "selfLink" : "/apis/stable.example.com/v1/crontabs/"
   }, 
   "items" : [
      {
         "spec" : {
            "image" : "old-image",
            "cronSpec" : "* * * * /5"
         }, 
         "apiVersion" : "stable.example.com/v1",
         "kind" : "CronTab",
         "metadata" : {
            "deletionGracePeriodSeconds" : null,
            "creationTimestamp" : "2017-07-06T18:52:03Z",
            "namespace" : "default",
            "uid" : "3385cb16-627c-11e7-9d68-a0481c9805b1",
            "deletionTimestamp" : null,
            "resourceVersion" : "1363",
            "clusterName" : "",
            "name" : "my-new-cron-object",
            "selfLink" : "/apis/stable.example.com/v1/namespaces/default/crontabs/my-new-cron-object"
         }
      },
      {
         "spec" : {
            "cronSpec" : "* * * * /5",
            "image" : "my-awesome-cron-image"
         },
         "apiVersion" : "stable.example.com/v1",
         "kind" : "CronTab",
         "metadata" : {
            "name" : "crontab-1",
            "clusterName" : "",
            "selfLink" : "/apis/stable.example.com/v1/namespaces/ns2/crontabs/crontab-1",
            "creationTimestamp" : "2017-07-06T18:56:00Z",
            "resourceVersion" : "1796",
            "deletionTimestamp" : null,
            "uid" : "c05f8fd5-627c-11e7-9d68-a0481c9805b1",
            "namespace" : "ns2",
            "deletionGracePeriodSeconds" : null
         }
      }
   ],
   "apiVersion" : "stable.example.com/v1"
}
```

**Comment by djmckinley** [comment]
@mbohlool - is this what you are looking for?

```
[root@ost-djm1-master56ey ~]# kubectl get tests --all-namespaces
NAMESPACE     NAME      KIND
default       test1     Test.v1.occloud.oracle.com
default       test2     Test.v1.occloud.oracle.com
mynamespace   test3     Test.v1.occloud.oracle.com
mynamespace   test4     Test.v1.occloud.oracle.com
[root@ost-djm1-master56ey ~]# kubectl get tests --all-namespaces -v9
I0707 00:55:42.756703   24270 loader.go:354] Config loaded from file /root/.kube/config
I0707 00:55:42.758361   24270 cached_discovery.go:118] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/servergroups.json
I0707 00:55:42.758508   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/authentication.k8s.io/v1/serverresources.json
I0707 00:55:42.758594   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/authentication.k8s.io/v1beta1/serverresources.json
I0707 00:55:42.758753   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/authorization.k8s.io/v1/serverresources.json
I0707 00:55:42.758841   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/authorization.k8s.io/v1beta1/serverresources.json
I0707 00:55:42.758913   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/autoscaling/v1/serverresources.json
I0707 00:55:42.758999   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/batch/v1/serverresources.json
I0707 00:55:42.759090   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/certificates.k8s.io/v1beta1/serverresources.json
I0707 00:55:42.759264   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/extensions/v1beta1/serverresources.json
I0707 00:55:42.759356   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/policy/v1beta1/serverresources.json
I0707 00:55:42.759439   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/rbac.authorization.k8s.io/v1beta1/serverresources.json
I0707 00:55:42.759518   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/rbac.authorization.k8s.io/v1alpha1/serverresources.json
I0707 00:55:42.759605   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/settings.k8s.io/v1alpha1/serverresources.json
I0707 00:55:42.759685   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/storage.k8s.io/v1beta1/serverresources.json
I0707 00:55:42.759756   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/storage.k8s.io/v1/serverresources.json
I0707 00:55:42.759976   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/apps/v1beta1/serverresources.json
I0707 00:55:42.760079   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/projectcalico.org/v1/serverresources.json
I0707 00:55:42.760143   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/oracle.com/v1/serverresources.json
I0707 00:55:42.760236   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/occloud.oracle.com/v1/serverresources.json
I0707 00:55:42.760476   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/v1/serverresources.json
I0707 00:55:42.760992   24270 cached_discovery.go:118] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/servergroups.json
I0707 00:55:42.761141   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/authentication.k8s.io/v1/serverresources.json
I0707 00:55:42.761262   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/authentication.k8s.io/v1beta1/serverresources.json
I0707 00:55:42.761364   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/authorization.k8s.io/v1/serverresources.json
I0707 00:55:42.761432   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/authorization.k8s.io/v1beta1/serverresources.json
I0707 00:55:42.761507   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/autoscaling/v1/serverresources.json
I0707 00:55:42.761626   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/batch/v1/serverresources.json
I0707 00:55:42.761710   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/certificates.k8s.io/v1beta1/serverresources.json
I0707 00:55:42.761848   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/extensions/v1beta1/serverresources.json
I0707 00:55:42.761912   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/policy/v1beta1/serverresources.json
I0707 00:55:42.761982   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/rbac.authorization.k8s.io/v1beta1/serverresources.json
I0707 00:55:42.762060   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/rbac.authorization.k8s.io/v1alpha1/serverresources.json
I0707 00:55:42.762123   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/settings.k8s.io/v1alpha1/serverresources.json
I0707 00:55:42.762196   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/storage.k8s.io/v1beta1/serverresources.json
I0707 00:55:42.762278   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/storage.k8s.io/v1/serverresources.json
I0707 00:55:42.762354   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/apps/v1beta1/serverresources.json
I0707 00:55:42.762412   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/projectcalico.org/v1/serverresources.json
I0707 00:55:42.762463   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/oracle.com/v1/serverresources.json
I0707 00:55:42.762529   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/occloud.oracle.com/v1/serverresources.json
I0707 00:55:42.762842   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/v1/serverresources.json
I0707 00:55:42.763145   24270 cached_discovery.go:118] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/servergroups.json
I0707 00:55:42.763224   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/authentication.k8s.io/v1/serverresources.json
I0707 00:55:42.763286   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/authentication.k8s.io/v1beta1/serverresources.json
I0707 00:55:42.763402   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/authorization.k8s.io/v1/serverresources.json
I0707 00:55:42.763490   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/authorization.k8s.io/v1beta1/serverresources.json
I0707 00:55:42.763570   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/autoscaling/v1/serverresources.json
I0707 00:55:42.763681   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/batch/v1/serverresources.json
I0707 00:55:42.763745   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/certificates.k8s.io/v1beta1/serverresources.json
I0707 00:55:42.763862   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/extensions/v1beta1/serverresources.json
I0707 00:55:42.763923   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/policy/v1beta1/serverresources.json
I0707 00:55:42.764002   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/rbac.authorization.k8s.io/v1beta1/serverresources.json
I0707 00:55:42.764078   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/rbac.authorization.k8s.io/v1alpha1/serverresources.json
I0707 00:55:42.764137   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/settings.k8s.io/v1alpha1/serverresources.json
I0707 00:55:42.764191   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/storage.k8s.io/v1beta1/serverresources.json
I0707 00:55:42.764286   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/storage.k8s.io/v1/serverresources.json
I0707 00:55:42.764367   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/apps/v1beta1/serverresources.json
I0707 00:55:42.764428   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/projectcalico.org/v1/serverresources.json
I0707 00:55:42.764482   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/oracle.com/v1/serverresources.json
I0707 00:55:42.764572   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/occloud.oracle.com/v1/serverresources.json
I0707 00:55:42.764847   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/v1/serverresources.json
I0707 00:55:42.765099   24270 cached_discovery.go:118] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/servergroups.json
I0707 00:55:42.765166   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/authentication.k8s.io/v1/serverresources.json
I0707 00:55:42.765267   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/authentication.k8s.io/v1beta1/serverresources.json
I0707 00:55:42.765339   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/authorization.k8s.io/v1/serverresources.json
I0707 00:55:42.765424   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/authorization.k8s.io/v1beta1/serverresources.json
I0707 00:55:42.765488   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/autoscaling/v1/serverresources.json
I0707 00:55:42.765564   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/batch/v1/serverresources.json
I0707 00:55:42.765636   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/certificates.k8s.io/v1beta1/serverresources.json
I0707 00:55:42.765767   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/extensions/v1beta1/serverresources.json
I0707 00:55:42.765851   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/policy/v1beta1/serverresources.json
I0707 00:55:42.765929   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/rbac.authorization.k8s.io/v1beta1/serverresources.json
I0707 00:55:42.766012   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/rbac.authorization.k8s.io/v1alpha1/serverresources.json
I0707 00:55:42.766084   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/settings.k8s.io/v1alpha1/serverresources.json
I0707 00:55:42.766150   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/storage.k8s.io/v1beta1/serverresources.json
I0707 00:55:42.766235   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/storage.k8s.io/v1/serverresources.json
I0707 00:55:42.766312   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/apps/v1beta1/serverresources.json
I0707 00:55:42.766378   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/projectcalico.org/v1/serverresources.json
I0707 00:55:42.766434   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/oracle.com/v1/serverresources.json
I0707 00:55:42.766505   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/occloud.oracle.com/v1/serverresources.json
I0707 00:55:42.766786   24270 cached_discovery.go:71] returning cached discovery info from /root/.kube/cache/discovery/172.31.0.1_443/v1/serverresources.json
I0707 00:55:42.772807   24270 round_trippers.go:398] curl -k -v -XGET  -H "Accept: application/json" -H "User-Agent: kubectl/v1.6.2 (linux/amd64) kubernetes/477efc3" https://172.31.0.1:443/apis/occloud.oracle.com/v1/tests
I0707 00:55:42.801595   24270 round_trippers.go:417] GET https://172.31.0.1:443/apis/occloud.oracle.com/v1/tests 200 OK in 28 milliseconds
I0707 00:55:42.801646   24270 round_trippers.go:423] Response Headers:
I0707 00:55:42.801654   24270 round_trippers.go:426]     Content-Type: application/json
I0707 00:55:42.801660   24270 round_trippers.go:426]     Content-Length: 1522
I0707 00:55:42.801676   24270 round_trippers.go:426]     Date: Fri, 07 Jul 2017 00:55:42 GMT
I0707 00:55:42.801726   24270 request.go:991] Response Body: {"kind":"TestList","items":[{"apiVersion":"occloud.oracle.com/v1","kind":"Test","metadata":{"name":"test1","namespace":"default","selfLink":"/apis/occloud.oracle.com/v1/namespaces/default/tests/test1","uid":"c699909a-6298-11e7-9dda-fa163e3667f1","resourceVersion":"2449709","creationTimestamp":"2017-07-06T22:16:36Z"},"spec":{"testCount":1,"testType":"test1 type"}},{"apiVersion":"occloud.oracle.com/v1","kind":"Test","metadata":{"name":"test2","namespace":"default","selfLink":"/apis/occloud.oracle.com/v1/namespaces/default/tests/test2","uid":"d8402752-6298-11e7-9dda-fa163e3667f1","resourceVersion":"2449765","creationTimestamp":"2017-07-06T22:17:06Z"},"spec":{"testCount":2,"testType":"test2 type"}},{"apiVersion":"occloud.oracle.com/v1","kind":"Test","metadata":{"name":"test3","namespace":"mynamespace","selfLink":"/apis/occloud.oracle.com/v1/namespaces/mynamespace/tests/test3","uid":"c1b8fa7a-629b-11e7-9dda-fa163e3667f1","resourceVersion":"2452133","creationTimestamp":"2017-07-06T22:37:56Z"},"spec":{"testCount":3,"testType":"test3 type"}},{"apiVersion":"occloud.oracle.com/v1","kind":"Test","metadata":{"name":"test4","namespace":"mynamespace","selfLink":"/apis/occloud.oracle.com/v1/namespaces/mynamespace/tests/test4","uid":"c3973d62-629b-11e7-9dda-fa163e3667f1","resourceVersion":"2452139","creationTimestamp":"2017-07-06T22:37:59Z"},"spec":{"testCount":4,"testType":"test4 type"}}],"metadata":{"selfLink":"/apis/occloud.oracle.com/v1/tests","resourceVersion":"2467731"},"apiVersion":"occloud.oracle.com/v1"}
NAMESPACE     NAME      KIND
default       test1     Test.v1.occloud.oracle.com
default       test2     Test.v1.occloud.oracle.com
mynamespace   test3     Test.v1.occloud.oracle.com
mynamespace   test4     Test.v1.occloud.oracle.com
```

**Comment by mbohlool** [comment]
I think you are hitting the bug that Anthony found. What we need to do is to merge this PR and update main repo's generated client and test your code against it.

**Comment by djmckinley** [comment]
Let me know when and where to clone the repo from and I'll try again.

**Comment by mbohlool** [comment]
`tpr` branch in my (mbohlool) clone of client-python repo. I reopened the same PR there.if you already cloned it, you need to remove branch and resync again as I force pushed new changes to that PR's branch.

**Comment by djmckinley** [comment]
Testing the latest commit on your tpr branch - it worked great!

```
bash-4.2# python3 lcr.py
configured via kubeconfig file

Calling list_namespaced_custom_object for 'default' namespace

{'apiVersion': 'occloud.oracle.com/v1',
 'items': [{'apiVersion': 'occloud.oracle.com/v1',
            'kind': 'Test',
            'metadata': {'creationTimestamp': '2017-07-06T22:16:36Z',
                         'name': 'test1',
                         'namespace': 'default',
                         'resourceVersion': '2449709',
                         'selfLink': '/apis/occloud.oracle.com/v1/namespaces/default/tests/test1',
                         'uid': 'c699909a-6298-11e7-9dda-fa163e3667f1'},
            'spec': {'testCount': 1, 'testType': 'test1 type'}},
           {'apiVersion': 'occloud.oracle.com/v1',
            'kind': 'Test',
            'metadata': {'creationTimestamp': '2017-07-06T22:17:06Z',
                         'name': 'test2',
                         'namespace': 'default',
                         'resourceVersion': '2449765',
                         'selfLink': '/apis/occloud.oracle.com/v1/namespaces/default/tests/test2',
                         'uid': 'd8402752-6298-11e7-9dda-fa163e3667f1'},
            'spec': {'testCount': 2, 'testType': 'test2 type'}}],
 'kind': 'TestList',
 'metadata': {'resourceVersion': '2555031',
              'selfLink': '/apis/occloud.oracle.com/v1/namespaces/default/tests'}}

Calling list_namespaced_custom_object for 'mynamespace' namespace

{'apiVersion': 'occloud.oracle.com/v1',
 'items': [{'apiVersion': 'occloud.oracle.com/v1',
            'kind': 'Test',
            'metadata': {'creationTimestamp': '2017-07-06T22:37:56Z',
                         'name': 'test3',
                         'namespace': 'mynamespace',
                         'resourceVersion': '2452133',
                         'selfLink': '/apis/occloud.oracle.com/v1/namespaces/mynamespace/tests/test3',
                         'uid': 'c1b8fa7a-629b-11e7-9dda-fa163e3667f1'},
            'spec': {'testCount': 3, 'testType': 'test3 type'}},
           {'apiVersion': 'occloud.oracle.com/v1',
            'kind': 'Test',
            'metadata': {'creationTimestamp': '2017-07-06T22:37:59Z',
                         'name': 'test4',
                         'namespace': 'mynamespace',
                         'resourceVersion': '2452139',
                         'selfLink': '/apis/occloud.oracle.com/v1/namespaces/mynamespace/tests/test4',
                         'uid': 'c3973d62-629b-11e7-9dda-fa163e3667f1'},
            'spec': {'testCount': 4, 'testType': 'test4 type'}}],
 'kind': 'TestList',
 'metadata': {'resourceVersion': '2555031',
              'selfLink': '/apis/occloud.oracle.com/v1/namespaces/mynamespace/tests'}}

Calling list_cluster_custom_object

{'apiVersion': 'occloud.oracle.com/v1',
 'items': [{'apiVersion': 'occloud.oracle.com/v1',
            'kind': 'Test',
            'metadata': {'creationTimestamp': '2017-07-06T22:16:36Z',
                         'name': 'test1',
                         'namespace': 'default',
                         'resourceVersion': '2449709',
                         'selfLink': '/apis/occloud.oracle.com/v1/namespaces/default/tests/test1',
                         'uid': 'c699909a-6298-11e7-9dda-fa163e3667f1'},
            'spec': {'testCount': 1, 'testType': 'test1 type'}},
           {'apiVersion': 'occloud.oracle.com/v1',
            'kind': 'Test',
            'metadata': {'creationTimestamp': '2017-07-06T22:17:06Z',
                         'name': 'test2',
                         'namespace': 'default',
                         'resourceVersion': '2449765',
                         'selfLink': '/apis/occloud.oracle.com/v1/namespaces/default/tests/test2',
                         'uid': 'd8402752-6298-11e7-9dda-fa163e3667f1'},
            'spec': {'testCount': 2, 'testType': 'test2 type'}},
           {'apiVersion': 'occloud.oracle.com/v1',
            'kind': 'Test',
            'metadata': {'creationTimestamp': '2017-07-06T22:37:56Z',
                         'name': 'test3',
                         'namespace': 'mynamespace',
                         'resourceVersion': '2452133',
                         'selfLink': '/apis/occloud.oracle.com/v1/namespaces/mynamespace/tests/test3',
                         'uid': 'c1b8fa7a-629b-11e7-9dda-fa163e3667f1'},
            'spec': {'testCount': 3, 'testType': 'test3 type'}},
           {'apiVersion': 'occloud.oracle.com/v1',
            'kind': 'Test',
            'metadata': {'creationTimestamp': '2017-07-06T22:37:59Z',
                         'name': 'test4',
                         'namespace': 'mynamespace',
                         'resourceVersion': '2452139',
                         'selfLink': '/apis/occloud.oracle.com/v1/namespaces/mynamespace/tests/test4',
                         'uid': 'c3973d62-629b-11e7-9dda-fa163e3667f1'},
            'spec': {'testCount': 4, 'testType': 'test4 type'}}],
 'kind': 'TestList',
 'metadata': {'resourceVersion': '2555031',
              'selfLink': '/apis/occloud.oracle.com/v1/tests'}}
```

**Pull request merged**